File: BackEnd/retail.demo/src/main/java/com/retail/demo/service/OrderedItemService.java
package com.retail.demo.service;

import com.retail.demo.model.OrderedItem;

import java.util.List;

public interface OrderedItemService {
    OrderedItem createOrderedItem(OrderedItem orderedItem);
    OrderedItem updateOrderedItem(Long id, OrderedItem orderedItem);
    OrderedItem getOrderedItem(Long id);
    boolean deleteOrderedItem(Long id);
    List<OrderedItem> getAllOrderedItemsFromOrder(Long orderId);
    List<OrderedItem> getAllOrderedItems();
    OrderedItem getLatestAddition();
    boolean isDaoEmpty();
}

File: FrontEnd/retail-web-app/src/service/OrderedItemService.js
import axios from 'axios';
const API_URL = 'http://localhost:9080/OrderedItemApi/v1/orderedItems';

class OrderedItemService {
    
        getAllOrderedItems() {
            return axios.get(API_URL);
        }
    
        getOrderedItemById(orderedItemId) {
            return axios.get(API_URL + '/' + orderedItemId);
        }
    
        getOrderedItemsByOrderId(orderId) {
            return axios.get(API_URL + '/order/' + orderId);
        }
    
        createOrderedItem(orderedItem) {
            return axios.post(API_URL, orderedItem);
        }
    
        updateOrderedItem(orderedItem, orderedItemId) {
            return axios.put(API_URL + '/' + orderedItemId, orderedItem);
        }
    
        deleteOrderedItem(orderedItemId) {
            return axios.delete(API_URL + '/' + orderedItemId);
        }
}

export default new OrderedItemService()